js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+            //     {
+            //          success: true,
+            //          message: "",
+            //          result: {
+            //            bid: '46504.7412435',
+            //            ask: '46511.54203562',
+            //            open: '47030.33',
+            //            high: '47549.99',
+            //            low: '45571.08',
+            //            last: '46459.99',
+            //            volume: '8372.29022355',
+            //            deal: '388090406.73607745',
+            //          },
+            //          code: "200"
+            //     }
+            ticker = this.safeValue (response, 'result', {});
+        } else if (exchangeType === 'futures') {
+            // data = await this.futuresPublicGetTicker (request);
+        }
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const [exchangeType, request] = this.checkExchangeType ('fetchTickers', params);
+        await this.loadMarkets ();
+        let tickers = {};
+        if (exchangeType === 'spot') {
+            const response = await this.spotPublicGetTickers (request);
+            //     {
+            //          success: true,
+            //          message: "",
+            //          result: {
+            //            BTC_USDT: {
+            //               at: '1640028811',
+            //               ticker: {
+            //                 bid: '46448.49682539',
+            //                 ask: '46458.41093569',
+            //                 open: '46687.39',
+            //                 high: '47550',
+            //                 low: '45584.27',
+            //                 last: '46462.77',
+            //                 vol: '32.56275846',
+            //                 deal: '1510574.62551143',
+            //                 change: '-0'
+            //               }
+            //            },
+            //            ...
+            //          },
+            //          code: "200"
+            //     }
+            tickers = this.safeValue (response, 'result', {});
+        } else if (exchangeType === 'futures') {
+            // tickers = ..
+        }
+        return this.parseTickers (tickers, undefined, params);
+    }
+
+    // This modified parseTickers method needs to exist in base, to keep the KYES & pass them through parseTickers
+    parseTickers (tickers, symbols = undefined, params = {}) {
+        const result = [];
+        const values = Object.values (tickers || []);
+        const keys = Object.keys (tickers);
+        for (let i = 0; i < values.length; i++) {
+            result.push (this.extend (this.parseTicker (values[i], undefined, keys[i]), params));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined, key = undefined) {
+        let tickerFinal = {};
```

Comment:
Ideally, we should write out the full structure we're parsing here, see how `parseTicker` is formatted in the other exchanges – that allows seeing what you're working with when writing the parsing code below.

Reply:
k